internal/reporter/checkstyle.go

```diff
@@ -0,0 +1,67 @@
+package reporter
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"strings"
+)
+
+func NewCheckStyleReporter(output io.Writer) CheckStyleReporter {
+	return CheckStyleReporter{
+		output: output,
+	}
+}
+
+type CheckStyleReporter struct {
+	output io.Writer
+}
+
+type Dirs map[string][]Report
+
+func sortByFile(summary Summary) Dirs {
+	x := make(Dirs)
+	for _, report := range summary.reports {
+		x[report.Path.Name] = append(x[report.Path.Name], report)
+	}
+	return x
+}
+
+func (cs CheckStyleReporter) Submit(summary Summary) error {
+	dirs := sortByFile(summary)
+	var buf strings.Builder
+	buf.WriteString("<?xml version='1.0' encoding='UTF-8'?>\n")
```

Comment:
`encoding/xml` can generate the entire XML file for you, that should be safer & more correct than manually crafting it.

Reply:
Sorry here I am not sure what you are referring to exactly. I see I could be using `xml.Header`.
Do you mean by having a custom struct that marshals to `<checkstyle version='4.3'>` making dirs marshal to `<file name=....` and report to <error line=.../>?